ostd/src/arch/x86/cpu.rs

```diff
@@ -292,6 +290,8 @@ impl UserContextApiInternal for UserContext {
         // Go back to user when it is syscall or cpu exception type is an immediately
         // recoverable Fault or Trap. Otherwise return to handle the exception.
         let return_reason = loop {
+            // Make sure that when running the user code, the local interrupt is enabled.
+            debug_assert!(crate::arch::irq::is_local_enabled());
```

Comment:
The modification to the RFlags has already enabled the interrupt in user mode. The CPU will load the RFlags from `self.user_context.general.rflags`. Therefore, there is no need to check whether the interrupt is enabled here.

Reply:
I introduced this check because I do not want the kernel mistakenly holding a irq guard before returning. Maybe it's not needed.

Reply:
I think it's okay to check the IRQ guard here. In that case, the comment should be changed.